src/openloco/windows/build_vehicle.cpp

```diff
@@ -122,4 +463,519 @@ namespace openloco::ui::build_vehicle
+
+        WindowManager::invalidateWidget(WindowType::buildVehicle, window->number, window->current_tab + 4);
+        WindowManager::invalidateWidget(WindowType::buildVehicle, window->number, (window->var_874 & 0xFF) + 10);
+        WindowManager::invalidateWidget(WindowType::buildVehicle, window->number, 19);
+    }
+
+    // 0x4C37B9
+    static void get_scroll_size(ui::window* window, uint32_t scrollIndex, uint16_t* scrollWidth, uint16_t* scrollHeight)
+    {
+        *scrollHeight = window->var_83C * window->row_height;
+    }
+
+    // 0x4C384B
+    static void on_scroll_mouse_down(ui::window* window, int16_t x, int16_t y, uint8_t scroll_index)
+    {
+        if (scroll_index != 0)
+        {
+            return;
+        }
+
+        auto scroll_item = y / window->row_height;
+        if (scroll_item >= window->var_83C)
+        {
+            return;
+        }
+
+        auto pan = window->width / 2 + window->x;
+        audio::play_sound(audio::sound_id::click_down, loc16{ x, y, static_cast<int16_t>(pan) }, pan);
+        auto item = window->row_info[scroll_item];
+        auto vehicle_obj = objectmgr::get<vehicle_object>(item);
+        _common_format_args[5] = vehicle_obj->name;
+        gGameCommandErrorTitle = string_ids::cant_build_pop_5_string_id;
+        if (_build_target_vehicle != -1)
+        {
+            auto vehicle = thingmgr::get<openloco::vehicle>(_build_target_vehicle);
+            _common_format_args[6] = vehicle->var_44;
+            _common_format_args[7] = vehicle->var_22;
+            gGameCommandErrorTitle = string_ids::cant_add_pop_5_string_id_string_id;
+        }
+
+        if (!game_commands::do_5(item, _build_target_vehicle))
+        {
+            return;
+        }
+
+        if (_build_target_vehicle == -1)
+        {
+            auto vehicle = thingmgr::get<openloco::vehicle>(_113642A);
+            sub_4B60CC(vehicle);
+        }
+        sub_4B92A5(window);
+    }
+
+    // 0x4C3802
+    static void on_scroll_mouse_over(ui::window* window, int16_t x, int16_t y, uint8_t scroll_index)
+    {
+        if (scroll_index != 0)
+        {
+            return;
+        }
+
+        auto scroll_item = y / window->row_height;
+        int16_t item = -1;
+        if (scroll_item < window->var_83C)
+        {
+            item = window->row_info[scroll_item];
+        }
+
+        if (item != -1 && item != window->row_hover)
+        {
+            window->row_hover = item;
+            window->invalidate();
+        }
+    }
+
+    // 0x4C370C
+    static void tooltip(ui::window* window, widget_index widgetIndex)
+    {
+        if (widgetIndex < widx::tab_vehicles_for_0 || widgetIndex >= widx::scrollview_1)
+        {
+            *_common_format_args = string_ids::tooltip_scroll_new_vehicle_list;
+        }
+        else
+        {
+            auto unk_for_tab = widget_index_to_tab_vehicle_for(widgetIndex);
+            auto type = _11364F0[unk_for_tab];
+            if (type == 0xFF)
```

Comment:
Linux and macOS CI are complaining here. `_11364F0` is defined as an `int8_t`, so 0xFF is out of range.
```suggestion
            if (type == -1)
```